src/schema/server_settings.py

```diff
@@ -48,3 +114,88 @@ def _select_model(
         raise SteamshipError(
             message=f"Invalid model selection (preferred={preferred}, default={default}). Only the following are allowed: {allowed}"
         )
+
+    def update_from_web(self, other: "ServerSettings"):  # noqa: C901
+        """Perform a gentle update so that the website doesn't accidentally blast over this if it diverges in
+        structure."""
```

Comment:
Frontend can 100% just pass back dirtied fields -- I think that's probably the preferred way to do it.

I've been wondering if we'll find we need some kind of `__DELETE__` magic value that means "zero out this field" since that would make it hard to delete a setting.. but in general I think the "only pass back the dirtied fields" approach feels better overall.